src/main/java/org/broadinstitute/consent/http/models/Dataset.java

```diff
@@ -297,6 +304,59 @@ public void setSharingPlanDocumentName(String sharingPlanDocumentName) {
         this.sharingPlanDocumentName = sharingPlanDocumentName;
     }
 
+    /**
+     * Checks if the Dataset matches a raw search query. Searches on all dataset properties
+     * and some data use properties.
+     *
+     * @param query Raw string query
+     * @return if the Dataset matched query
+     */
+    public boolean isStringMatch(@NonNull String query) {
+        String lowerCaseQuery = query.toLowerCase();
+        List<String> queryTerms = new ArrayList<>(List.of(lowerCaseQuery.split("\\s+")));
+
+        List<String> matchTerms = new ArrayList<>();
+        matchTerms.add(this.getName());
+        matchTerms.add(this.getDatasetIdentifier());
+
+        if (Objects.nonNull(this.getProperties())) {
+            this.getProperties().forEach(prop -> {
+                if (Objects.nonNull(prop.getPropertyValue())) {
+                    matchTerms.add(prop.getPropertyValueAsString().toLowerCase());
+                }
+            });
+        }
+
+        if (Objects.nonNull(this.dataUse)) {
+            if (Objects.nonNull(this.dataUse.getEthicsApprovalRequired())
+                    && this.dataUse.getEthicsApprovalRequired()) {
+                matchTerms.add("irb");
+            }
+
+            if (Objects.nonNull(this.dataUse.getCollaboratorRequired())
+                    && this.dataUse.getCollaboratorRequired()) {
+                matchTerms.add("collaborator");
+            }
+
+            if (Objects.nonNull(this.dataUse.getDiseaseRestrictions())) {
+                matchTerms.addAll(this.dataUse.getDiseaseRestrictions());
+            }
+        }
```

Comment:
What would be really nice here is the ability to pull a full string from a `DataUse` that has all of these values. We do have that, but not here - it's in Ontology:
```
curl -X 'POST' 'https://consent-ontology.dsde-prod.broadinstitute.org/translate?for=dataset' -H 'accept: text/plain' -H 'Content-Type: application/json' -d '{ "generalUse": false, "hmbResearch": true, "diseaseRestrictions": [ "http://purl.obolibrary.org/obo/DOID_602" ] }'

Samples are restricted for use under the following conditions:
Data is limited for health/medical/biomedical research. [HMB]
Data use is limited for studying: cancerophobia [DS]
Commercial use is not prohibited.
Data use for methods development research irrespective of the specified data use limitations is not prohibited.
Restrictions for use as a control set for diseases other than those defined were not specified.
```
Maybe as a future feature, we might have this saved as part of the dataset itself which would facilitate this matching better.

Reply:
I agree with this, I was searching for something like this originally and since I couldn't find it ultimately abandoned the idea of searching on all data use terms (since it would require many lines of code which would require frequent updating). Saving this would be very helpful I think.